source/extensions/common/wasm/wasm.cc

```diff
@@ -243,18 +256,20 @@ void Wasm::getFunctions() {
   }
 }
 
-Wasm::Wasm(const Wasm& wasm, Event::Dispatcher& dispatcher)
-    : std::enable_shared_from_this<Wasm>(wasm), vm_id_(wasm.vm_id_),
-      vm_id_with_hash_(wasm.vm_id_with_hash_), started_from_(wasm.wasm_vm()->cloneable()),
-      scope_(wasm.scope_), cluster_manager_(wasm.cluster_manager_), dispatcher_(dispatcher),
-      time_source_(dispatcher.timeSource()), wasm_stats_(wasm.wasm_stats_),
-      stat_name_set_(wasm.stat_name_set_) {
+Wasm::Wasm(WasmHandleSharedPtr& wasm_handle, Event::Dispatcher& dispatcher)
+    : std::enable_shared_from_this<Wasm>(*wasm_handle->wasm()), vm_id_(wasm_handle->wasm()->vm_id_),
+      vm_key_(wasm_handle->wasm()->vm_key_),
+      started_from_(wasm_handle->wasm()->wasm_vm()->cloneable()),
+      scope_(wasm_handle->wasm()->scope_), cluster_manager_(wasm_handle->wasm()->cluster_manager_),
+      dispatcher_(dispatcher), time_source_(dispatcher.timeSource()),
+      base_wasm_handle_(wasm_handle), wasm_stats_(wasm_handle->wasm()->wasm_stats_),
+      stat_name_set_(wasm_handle->wasm()->stat_name_set_) {
   if (started_from_ != Cloneable::NotCloneable) {
-    wasm_vm_ = wasm.wasm_vm()->clone();
+    wasm_vm_ = wasm_handle->wasm()->wasm_vm()->clone();
```

Comment:
Nit: Use `base_wasm_handle_` instead of `wasm_handle` (I think it's better for readability, but feel free to ignore this).

Reply:
Done